shard.go

```diff
@@ -234,7 +234,9 @@ func (s *Shard) close() error {
 	if s.store != nil {
 		_ = s.store.Close()
 	}
-	s.stats.Inc("close")
```

Comment:
Also caused panic. We need to continue to rework the shard type in our code. @corylanou started some of this work.